### PR TITLE
refact(errors): add Cause() to custom errors package

### LINKED ISSUE #1134: Missing Cause() in new errors package
The new errors package does not implement the Cause() of the standard errors package which makes it difficult to implement it to old files.

Add the Cause() to the new errors package.

### PR DESCRIPTION
**Pull request by shubham14bajpai**
- added `Cause()` to the custom errors package

Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #1134 )` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/errors/v1alpha1/errors.go [hunk 1]
```diff
@@ -71,3 +71,19 @@ func WithStack(err error) error {
 		callers(),
 	}
 }
+
+// Cause returns the underlying cause of the error, if possible.
```
Comment (imazik) [inline pkg/errors/v1alpha1/errors.go]:
```
Cause returns the underlying cause of the error, if present.
else returns nil.
```

Comment (shubham14bajpai) [reply pkg/errors/v1alpha1/errors.go]: Done

File: pkg/errors/v1alpha1/types.go [hunk 2]
```diff
@@ -83,6 +83,9 @@ type wrapper struct {
 // Error is implementation of error interface
 func (w *wrapper) Error() string { return w.msg }
 
+// Cause is implementation of causer interface
+func (w *wrapper) Cause() error { return w.error }
```
Comment (AmitKumarDas) [inline pkg/errors/v1alpha1/types.go]: Do we need to implement Cause for other error based structures?

Comment (shubham14bajpai) [reply pkg/errors/v1alpha1/types.go]: I guess we don't need to as not all structures have a `error` type field. At most we can implement for `withSatck`.

**Comment by imazik** [comment]
@shubham14bajpai Please add `Cause()` for `WithStack`

**Comment by shubham14bajpai** [comment]
> @shubham14bajpai Please add `Cause()` for `WithStack`

Done

File: pkg/errors/v1alpha1/errors.go [hunk 3]
```diff
@@ -71,3 +71,20 @@ func WithStack(err error) error {
 		callers(),
 	}
 }
+
+// Cause returns the underlying cause of the error, if present.
```
Comment (sagarkrsd) [inline pkg/errors/v1alpha1/errors.go]: We can have the comments more descriptive to have a better understanding of this,
have picked this from errors package, see if it is better -
```
// Cause returns the underlying cause of the error, if possible.
// An error value has a cause if it implements the following
// interface:
//
//     type causer interface {
//            Cause() error
//     }
//
// If the error does not implement Cause, the original error will
// be returned. If the error is nil, nil will be returned without further
// investigation.
```

Comment (shubham14bajpai) [reply pkg/errors/v1alpha1/errors.go]: Done

**Pull request merged**